src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +672,28 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getCanonicalName() + ".generateOnOfflineNode")) {
```

Comment:
Somehow NIT: Though in that case `SupportPlugin.class.getCanonicalName()` will yield the same result as getName(), I'd use getName() since pretty much every calls with that pattern in the core use getName().